test/libyul/yulOptimizerTests/expressionInliner/argument_duplication_heuristic.yul

```diff
@@ -0,0 +1,39 @@
+{
+    // references parameter 6 times in function body
+    function ref6(a) -> x { x := add(a, sub(a, mul(a, div(a, mod(a, a))))) }
+    // references parameter 5 times in function body
+    function ref5(a) -> x { x := add(a, sub(a, mul(a, div(a, a)))) }
+    let y1 := ref6(calldatasize())
+    let y2 := ref5(calldatasize())
+    let y3 := ref6(0xff)
+    let y4 := ref5(0xff)
+    let y5 := ref6(0x123)
+    let y6 := ref5(0x123)
+    let y7 := ref6(add(1, 2))
+    let y8 := ref5(add(1, 2))
+    let y9 := ref6(y1)
+    let y10:= ref5(y1)
+}
+// ====
+// step: expressionInliner
+// ----
+// {
+//     function ref6(a) -> x
+//     {
+//         x := add(a, sub(a, mul(a, div(a, mod(a, a)))))
+//     }
+//     function ref5(a_1) -> x_2
+//     {
+//         x_2 := add(a_1, sub(a_1, mul(a_1, div(a_1, a_1))))
+//     }
+//     let y1 := add(calldatasize(), sub(calldatasize(), mul(calldatasize(), div(calldatasize(), mod(calldatasize(), calldatasize())))))
+//     let y2 := add(calldatasize(), sub(calldatasize(), mul(calldatasize(), div(calldatasize(), calldatasize()))))
+//     let y3 := ref6(0xff)
+//     let y4 := add(0xff, sub(0xff, mul(0xff, div(0xff, 0xff))))
+//     let y5 := ref6(0x123)
+//     let y6 := ref5(0x123)
+//     let y7 := ref6(add(1, 2))
+//     let y8 := ref5(add(1, 2))
+//     let y9 := ref6(y1)
```

Comment:
Is this desired behavior? In this case, refs = 6, cost = 1, thus the function is not inlined

Reply:
It's the metric of the rematerializer, which may not be the best here. Maybe we can relax to say cost<=1 is fine.

Reply:
Actually, we should not only take a look at the cost of arguments, but also at the size of the function to be inlined. If arguments are variables, they should not count negative because they will just be replaced by another variable.

Reply:
Then should this relaxation also apply to rematerializer? for example, in the following code, rematerializer won't replace `a` the last statement with `x`
```
{
    let x := add(1, 2)
    let b := x
    let a := x
    mstore(1, add(a, add(a, add(a, add(a, add(a, a))))))
}
```

Reply:
> Actually, we should not only take a look at the cost of arguments, but also at the size of the function to be inlined.

Ok but I guess this problem is not specific to `ExpressionInliner`? Not sure how does `FullInliner` handle this at this moment

Reply:
The expression inliner generates less overhead and single-expresion functions can more easily be simplified after inlining.

Reply:
About the rematerializer: I think because of the way it is currently invoked in the suite, it will not be applied to the input you gave.

Reply:
So the rule should be `refs <= 1 || cost <= 1`. Is this correct?

Reply:
Yes, I think that would work quite well.